### ISSUE TITLE
Cannot make a request to https endpoint

### ISSUE THREAD

**Issue by qezz** [paritytech/jsonrpsee#748]
I'm trying to make a request to an HTTPs endpoint, but it fails with a weird error

```rust
use jsonrpsee::core::client::ClientT;
use jsonrpsee::http_client::HttpClientBuilder;

pub async fn http_call() -> anyhow::Result<()> {
    let client = HttpClientBuilder::default()
        .build("https://kusama-rpc.polkadot.io:443")
        .unwrap();
    let response: String = client.request("system_chain", None).await.unwrap();

    tracing::info!("response: {:?}", response);

    Ok(())
}

#[tokio::main]
async fn main() {
    tracing_subscriber::FmtSubscriber::builder()
	.with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
	.try_init()
	.expect("setting default subscriber failed");

    http_call().await.unwrap();
}

```

output
```console
$ RUST_LOG=debug ./target/debug/sample
Apr 30 04:38:06.331 DEBUG hyper_rustls::config: with_native_roots processed 161 valid and 0 invalid certs
Apr 30 04:38:06.334 DEBUG jsonrpsee_http_client::transport: send: {"jsonrpc":"2.0","id":0,"method":"system_chain"}
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Transport(HTTP error: error trying to connect: invalid URL, scheme is not http)', src/main.rs:8:71
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

deps
```toml
tokio = { version = "1", features = ["full"] }
anyhow = "1.0.57"
tracing = "0.1.34"
tracing-subscriber = "0.2.0"
jsonrpsee = { version = "=0.11.0", features = ["full"] }
```

**Comment by niklasad1** [comment]
right, thanks,

the HTTP connector is reused for both http or https I thought it could be reused.

**Comment by niklasad1** [comment]
#750 is the fix... should be in next week I guess.

**Comment by qezz** [comment]
Thank you!

**Comment by niklasad1** [comment]
Fixed by #750
